services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -0,0 +1,124 @@
+
+        }
+    }
+
+    def "Creating a flow should create new meters on a switch"() {
+        given: "A switch with OpenFlow 1.3 support"
+        def sw = topology.getActiveSwitches().find { it.ofVersion == "OF_13" }
+
+        when: "Get default meters from the switch"
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+        assert defaultMeters
+
+        and: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+
+        then: "New meters should appear after flow setup"
+        def newMeters = northboundService.getAllMeters(sw.dpId)
+        def newMeterEntries = newMeters.meterEntries.findAll { !defaultMeters.meterEntries.contains(it) }
+        newMeterEntries.size() == 2
+
+        and: "All new meters should have KBPS, BURST and STATS flags installed"
+        newMeterEntries*.flags.every { it.size() == 3 }
+        newMeterEntries.every { it.flags.sort().equals(["KBPS", "BURST", "STATS"].sort()) }
+
+        and: "All new meters rate should be equal to flow's rate"
+        newMeterEntries*.rate.every { it == flow.maximumBandwidth }
+
+        and: "Delete a flow"
+        flowHelper.deleteFlow(flow.id)
+
+        and: "New meters should disappear from the switch"
+        def newestMeters = northboundService.getAllMeters(sw.dpId)
+        newestMeters.meterEntries.containsAll(defaultMeters.meterEntries)
+        newestMeters.meterEntries.size() == defaultMeters.meterEntries.size()
+    }
+
+    @Ignore
+    @Issue("https://github.com/telstra/open-kilda/issues/1673")
+    def "Meter burst size should be #burstSize kbps if the flow rate is #flowRate kbps"() {
+        setup: "A flow with #flowRate kbps bandwidth is created on OpenFlow 1.3 compatible switch"
+        def sw = topology.getActiveSwitches().find { it.ofVersion == "OF_13" }
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+        def flowPayload = flowHelper.singleSwitchFlow(sw)
+        flowPayload.setMaximumBandwidth(10)
+        def flow = flowHelper.addFlow(flowPayload)
+
+        when: "Update flow bandwidth to #flowRate kbps"
+        flow = northboundService.getFlow(flow.id)
+        flow.setMaximumBandwidth(flowRate)
+        northboundService.updateFlow(flow.id, flow)
+        sleep((Constants.RULES_DELETION_TIME + Constants.RULES_INSTALLATION_TIME) * 1000)
+
+        then: "New meters should be installed on the switch"
+        def newMeters = northboundService.getAllMeters(sw.dpId).meterEntries.findAll {
+            !defaultMeters.meterEntries.contains(it)
+        }
+        newMeters.size() == 2
+
+        and: "New meters rate should be equal to flow bandwidth"
+        newMeters*.rate.every { it == flowRate }
+
+        and: "New meters burst size should be 10% of the flow rate, but no less than 1 mbit/s"
+        newMeters*.burstSize.every { it == burstSize }
+
+        cleanup: "Delete the flow"
```

Comment:
We have a convention not to use such cleanups. Please move it under 'and' block.

Reply:
It can't be moved to "and" here as it needs to be cleaned up only once.